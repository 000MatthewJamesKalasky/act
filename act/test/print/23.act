namespace foo {

export 
template<pint W>
function tst (int x) : int<W>
{
  chp {
    self := x + 1
  }
}

}


defproc test()
{
   int x;
   chp {
      x := 1;
      x := foo::tst<32>(x)
    }
}

test t;
