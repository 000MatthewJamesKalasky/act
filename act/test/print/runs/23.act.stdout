namespace foo {
export function tst_332_4 (int<32> x) : int<32>;

export function tst_332_4 (int<32> x) : int<32>
{

/* instances */

/* connections */
chp {
self:=x+0x1
}
}


/* instances */

/* connections */
}
defproc test ();

defproc test ()
{

/* instances */
int<32> x;

/* connections */
chp {
x:=0x1;x:=::foo::tst_332_4(x)
}
}


/* instances */
test t;

/* connections */
